doc/ntp/ntp-design.md

```diff
@@ -0,0 +1,898 @@
+    prefix ntp;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-mgmt_vrf {
+        prefix mvrf;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+    }
+
+    import sonic-loopback-interface {
+        prefix lo;
+    }
+
+    import sonic-mgmt_port {
+        prefix mprt;
+    }
+
+    description
+        "NTP yang Module for SONiC OS";
+
+    revision 2021-04-07 {
+        description
+            "First revision";
+    }
+
+    typedef state {
+        description "State of the config";
+        type enumeration {
+            enum enabled;
+            enum disabled;
+        }
+    }
+
+    typedef yes-no {
+        description "Yes/No configuration";
+        type enumeration {
+            enum yes;
+            enum no;
+        }
+    }
+
+    typedef on-off {
+        description "On/Off configuration";
+        type enumeration {
+            enum yes;
+            enum no;
+        }
+    }
+
+    typedef ntp-key-type {
+        description "NTP key encrypt type";
+        type enumeration {
+            enum md5;
+            enum sda1;
+        }
+    }
+
+    typedef association-type {
+        description "NTP server association type";
+        type enumeration {
+            enum server;
+            enum peer;
+            enum pool;
+        }
+    }
+
+    container sonic-ntp {
+
+        container NTP {
+
+            container global {
+
+                description "Global NTP part of config_db.json";
+
+                leaf-list src_intf {
+                    type union {
+                        type leafref {
+                            path /port:sonic-port/port:PORT/port:PORT_LIST/port:name;
+                        }
+                        type leafref {
+                            path /lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:name;
+                        }
+                        type leafref {
+                            path /lo:sonic-loopback-interface/lo:LOOPBACK_INTERFACE/lo:LOOPBACK_INTERFACE_LIST/lo:name;
+                        }
+                        type leafref {
+                            path /mprt:sonic-mgmt_port/mprt:MGMT_PORT/mprt:MGMT_PORT_LIST/mprt:name;
+                        }
+                    }
+
+                }
+
+                leaf vrf {
+                    must "(current() != 'mgmt') or (/mvrf:sonic-mgmt_vrf/mvrf:MGMT_VRF_CONFIG/mvrf:vrf_global/mvrf:mgmtVrfEnabled = 'true')" {
+                        error-message "Must condition not satisfied. Try enable Management VRF.";
+                    }
+
+                    type string {
+                        pattern "mgmt|default";
+                    }
+
+                    description
+                        "NTP can be enabled only in one VRF at time. In this revision, it is either default VRF or Management VRF.";
+                }
+
+                leaf state {
```

Comment:
state is generally a read-only field, suggest to rename the field name as admin_state and make use of the existing typedefs from the sonic types YANG.

Reply:
ok